src/dune_rules/merlin_ident.ml

```diff
@@ -9,9 +9,13 @@ let for_lib l = Lib l
 
 let for_exes ~names = Exes names
 
+(* For debug purposes we use the name of one library or executable and the hash
+   of the others if there are multiple executables to name the merlin file *)
 let to_string = function
   | Lib name -> sprintf "lib-%s" (Lib_name.to_string name)
-  | Exes names -> sprintf "exe-%s" (String.concat ~sep:"-" names)
+  | Exes [ name ] -> sprintf "exe-%s" name
+  | Exes (name :: names) -> sprintf "exe-%s-%i" name (Hashtbl.hash names)
```

Comment:
What about using `Digest`? I'd prefer not to rely on OCaml's hashing implementation for this.